third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2471,10 +2471,9 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                 vec![global_access.into_exp()],
             );
         }
-
         self.error(
             loc,
-            &format!("undeclared `{}`", global_var_sym.display(self.env())),
+            &format!("undeclared `{}`", global_var_sym.display_simple(self.env())),
```

Comment:
You can't do it like this. What if the user actually provided a qualified name (see line 2392)?

Reply:
Added a check in this commit: https://github.com/aptos-labs/aptos-core/pull/12316/commits/aaacb5eade784c36a3f133e9f16a4b5b9b913992